src/main/java/at/hannibal2/skyhanni/features/nether/VolcanoExplosivityDisplay.kt

```diff
@@ -0,0 +1,44 @@
+package at.hannibal2.skyhanni.features.nether
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.TabListUpdateEvent
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.RenderUtils.renderString
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class VolcanoExplosivityDisplay {
+    private val config get() = SkyHanniMod.feature.crimsonIsle
+    private val patternGroup = RepoPattern.group("crimson.volcano")
+    private val headerPattern by patternGroup.pattern(
+        "header.tablistline",
+        "(?:§.)*Volcano Explosivity:(?:[\\S ]+)*"
+    )
+    private val statusPattern by patternGroup.pattern(
+        "status.tablistline",
+        " *(?<status>(?:§.)*INACTIVE)"
+    )
+    private var display = ""
+
+    @SubscribeEvent
+    fun onTick(event: TabListUpdateEvent) {
+        if (!isEnabled()) return
+        val tabList = event.tabList
+        if (tabList.none { headerPattern.matches(it) }) return
+        statusPattern.matchMatcher(tabList[tabList.indexOfFirst { headerPattern.matches(it) } + 1]) {
```

Comment:
Currently you iterate over `tabList` twice. Also if the the last line of the tablist matches the header this crashes.

```suggestion
        val headerIndex = tabList.indexOfFirst { headerPattern.matches(it) }
        if (headerIndex < 0 || headerIndex + 1 !in tabList.indices) return
        statusPattern.matchMatcher(tabList[headerIndex + 1]) {
```